addons/fire/functions/fnc_burnEffects.sqf

```diff
@@ -0,0 +1,191 @@
+    _lightFlare setLightIntensity 0;
+    _lightFlare setLightColor [1, 0.8, 0.8];
+    _lightFlare setLightUseFlare true;
+    _lightFlare setLightFlareMaxDistance 100;
+    _lightFlare setLightFlareSize 0;
+
+    if (_unit != ACE_player) then {
+        private _relativeAttachPoint = vectorNormalized (_unit worldToModelVisual (getPos ACE_player));
+        _relativeAttachPoint set [2, 0.5];
+        _lightFlare attachTo [_unit, _relativeAttachPoint];
+    } else {
+        _lightFlare attachTo [_unit, [0, 0, 0.3]];
+    };
+};
+
+private _fireSound = objNull;
+
+if (isServer) then {
+    _fireSound = createSoundSource ["Sound_Fire", _unitPos, [], 0];
+    _fireSound attachTo [_unit, [0, 0, 0], "Head"];
+};
+
+[{
+    params ["_args", "_pfhID"];
+    _args params ["_unit", "_fireParticle", "_smokeParticle", "_fireLight", "_lightFlare", "_fireSound"];
+
+    if (isNull _unit || {!(_unit call FUNC(isBurning))}) exitWith {
+        _pfhID call CBA_fnc_removePerFrameHandler;
+
+        deleteVehicle _fireParticle;
+        deleteVehicle _smokeParticle;
+        deleteVehicle _fireLight;
+        deleteVehicle _lightFlare;
+        deleteVehicle _fireSound;
+    };
+
+    // Display burn indicators
+    if (_unit == ACE_player && {alive _unit} && {isNil {_unit getVariable QGVAR(burnUIPFH)}}) then { // this accounts for player remote controlled a new unit
+        private _burnIndicatorPFH = [LINKFUNC(burnIndicator), 1, _unit] call CBA_fnc_addPerFrameHandler;
+        _unit setVariable [QGVAR(burnUIPFH), _burnIndicatorPFH];
+    };
+
+    if (!hasInterface) exitWith {};
+
+    private _intensity = _unit getVariable [QGVAR(intensity), 0];
+
+    _fireParticle setDropInterval (0.01 max linearConversion [BURN_MAX_INTENSITY, BURN_MIN_INTENSITY, _intensity, 0.03, 0.1, false]);
+    _fireParticle setParticleParams [
+        ["\A3\data_f\ParticleEffects\Universal\Universal", 16, 10, 32], // sprite sheet values
+        "", // animation name
+        "Billboard", // particle type
+        1, // timer period
+        0.7, // lifetime
+        "destructionEffect2", // position
+        [0, 0, 1], // move velocity
+        0, // rotation velocity
+        10, // weight
+        7.9, // volume
+        1, // rubbing
+        [0.3, 0.3], // size
+        [
+            [1, 1, 1, -0],
+            [1, 1, 1, -1],
+            [1, 1, 1, -1],
+            [1, 1, 1, -1],
+            [1, 1, 1, -0]
+        ], // colour
+        [0.5, 1], // animation speed
+        1, // random dir period
+        0, // random dir intensity
+        "", // on timer script
+        "", // before destroy script
+        _unit, // particle source
+        0,
+        false,
+        0,
+        [[0.8, 0.6, 0.2, 1]] // emissive color
+    ];
+    _fireParticle setParticleRandom [
+        0.04 * _intensity, // life time
+        [0.05, 0.05, 2], // position
+        [0.05 * _intensity, 0.05 * _intensity, 0.05 * _intensity], // move velocity
```

Comment:
I think this identical triple work should be done only once with a var.

Reply:
#9757